libs/langchain/langchain/retrievers/self_query/supabase.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Dict, Tuple
+
+from langchain.chains.query_constructor.ir import (
+    Comparator,
+    Comparison,
+    Operation,
+    Operator,
+    StructuredQuery,
+    Visitor,
+)
+
+
+class SupabaseVectorTranslator(Visitor):
+    """Translate Langchain filters to Supabase PostgREST filters."""
+
+    allowed_operators = [Operator.AND, Operator.OR]
+    """Subset of allowed logical operators."""
+    allowed_comparators = [
+        Comparator.EQ,
+        Comparator.NE,
+        Comparator.GT,
+        Comparator.GTE,
+        Comparator.LT,
+        Comparator.LTE,
+        Comparator.CONTAIN,
+        Comparator.LIKE,
+    ]
+    """Subset of allowed logical comparators."""
+
+    metadata_column = "metadata"
+
+    def _format_value(self, value: Any, comparator: Comparator) -> str:
+        if comparator == Comparator.CONTAIN:
+            return f"%{value}%"
+
+        return value
+
+    def _map_comparator(self, comparator: Comparator) -> str:
+        """
+        Maps Langchain comparator to PostgREST comparator:
+
+        https://postgrest.org/en/stable/references/api/tables_views.html#operators
+        """
+        return {
+            Comparator.EQ: "eq",
+            Comparator.NE: "neq",
+            Comparator.GT: "gt",
+            Comparator.GTE: "gte",
+            Comparator.LT: "lt",
+            Comparator.LTE: "lte",
+            Comparator.CONTAIN: "like",
```

Comment:
double checking, this is intentionally like right?

Reply:
Yeah this was intentional (but possibly misused). Noticed that the LLM preferred using `CONTAIN` with natural language queries, but the usage was mostly aka to `LIKE '%value%`. So added a special case that wraps `CONTAIN` values with `%`.

Reply:
the prompt will actually only specify the comparators in `allowed_comparators` (L19) as being options, so if you remove CONTAIN there then the model won't (or shouldn't) be using it. would that be a clearer solution?

Reply:
Yeah we can do that. When I tried excluding `CONTAIN` earlier, the LLM ended up choosing `LIKE` as expected, but without any wildcard symbols like `%` or `*`, so ended up being used as `EQ`. I imagine this is the case across all integration though, so can potentially be improved separately in another PR.

Will remove `CONTAIN` for now!